source/jquery.flot.touch.js

```diff
@@ -275,10 +271,12 @@
             return false;
         }
 
-        function preventEventPropagation(e) {
-            if (!gestureState.allowEventPropagation) {
+        function preventEventDefault(e) {
+            if (!gestureState.isUnsupportedGesture) {
                 e.preventDefault();
-                e.stopPropagation();
+                if (!plot.getOptions().propagateOriginTouch) {
```

Comment:
I think this condition can become:

` if ((e.touches.length === 1 && options.pan.enableTouch)`
`  || (e.touches.length === 2 && options.zoom.enableTouch))`

This should equate to if a user has pan enabled and does a pan gesture we will stop the propagation of the event, and will similarly do so when a user has zoom enabled and performs a pinch gesture.

The extra `propagateOriginTouch` option can present a confusing state where a user can still propagate the event despite it being handled. I don't see a good reason to allow this.

Reply:
I am afraid that this way would not meet our expectation. We'd like to enable pan, but disable zoom(pinch). In this case, the first `touchstart` event and the last `touchend` of a pinch interaction will be stopped from propagation, because these events have only one touch (the second touch does not start or already ends). However, the gesture reconizer (hammerjs) we used outside need all of that events to reconize the actual gesture. I have tried and found out the outside hammer fails.

By the way, complectly or partially stopping propagation of the origin events by a gesture reconizer seems not very common (while preventing default is common). I tested hammerjs that it do not stop any origin events even the events are handled as a gesture.

Again, my suggestion is to remove the stopping-event-propagation behavior at all, at lease to have an option to complectly disabled it.

Reply:
That's fine. Can we rename the option to `propagateSupportedGestures`?

Reply:
I feel it a little misleading to name this option with `gesture`. It controls the behavior of the origin touch events like `touchstart` and `touchmove`, while I will understand `gesture` as `pinchmove` or `doubletap`. `propagateSupportedGesture` sounds like it will allow the gesture event (e.g. `pinchmove` to bubble.

Anyway, I can rename it as your comments if you insist.